tensorbay/opendataset/HKD/loader.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT Licence.
+#
+# pylint: disable=invalid-name
+
+"""Dataloader for Finger Tapping Hand Keypoints Dataset(HKD)."""
+import csv
+import os
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.geometry import Keypoint2D
+from tensorbay.label import LabeledKeypoints2D
+from tensorbay.utility import chunked
+
+DATASET_NAME = "HKD"
+_CSV_NAMES = {
+    1: "subject1_fingercount_2D_Annotations_cropped",
+    2: "subject2_fingercount_2D_Annotations_cropped",
+    3: "fingerappose_subject3_2D_Annotations_cropped",
+    4: "subject4_2D_Annotations_cropped",
+}
+_IMAGE_NAMES = {
+    1: "subject1_fingercount_cropframe_",
+    2: "subject2_fingercount_cropframe_",
+    3: "fingerappose_subject3_cropframe_",
+    4: "subject4_cropframe_",
+}
+
+
+def HKD(path: str) -> Dataset:
+    """`HKD <http://vlm1.uta.edu/~srujana/HandPoseDataset/HKD_Cropped/>`_ dataset.
+
+    The file structure should be like:
+
+        <path>
+            AnnotatedData_subject1/
+                CropImages/
+                    subject1_fingercount_cropframe_2.jpg
+                    subject1_fingercount_cropframe_3.jpg
+                    ...
+                    subject1_fingercount_cropframe_210.jpg
+                subject1_fingercount_2D_Annotations_cropped.csv
+
+            AnnotatedData_subject2/
+                CropImages/
+                    subject2_fingercount_cropframe_2.jpg
+                    subject2_fingercount_cropframe_3.jpg
+                    ...
+                    subject2_fingercount_cropframe_207.jpg
+                subject2_fingercount_2D_Annotations_cropped.csv
+
+            AnnotatedData_subject3/
+                CropImages/
+                    fingerappose_subject3_cropframe_2.jpg
+                    fingerappose_subject3_cropframe_3.jpg
+                    ...
+                    fingerappose_subject3_cropframe_235.jpg
+                fingerappose_subject3_2D_Annotations_cropped.csv
+
+            AnnotatedData_subject4/
+                CropImages/
+                    subject4_cropframe_2.jpg
+                    subject4_cropframe_3.jpg
+                    ...
+                    subject4_cropframe_147.jpg
+                subject4_2D_Annotations_cropped.csv
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    for subject_id in range(1, 5):
```

Comment:
where the number `5` come from?
should we use the length of `_CSV_NAMES` instead?